src/linker/Linker.Steps/RemoveUnreachableBlocksStep.cs

```diff
@@ -177,16 +177,31 @@ bool TryInlineBodyDependencies (ref BodyReducer reducer)
 					if (!constExprMethods.TryGetValue (md, out targetResult))
 						break;
 
+					bool explicitlyAnnotated = Annotations.GetAction (md) == MethodAction.ConvertToStub;
+
 					// Allow inlining results of instance methods which are explicitly annotated
 					// but don't allow inling results of any other instance method.
 					// See https://github.com/mono/linker/issues/1243 for discussion as to why.
 					// Also explicitly prevent inlining results of virtual methods.
 					if (!md.IsStatic &&
-						(md.IsVirtual || Annotations.GetAction (md) != MethodAction.ConvertToStub))
+						(md.IsVirtual || !explicitlyAnnotated))
```

Comment:
Please add check for `vararg` as well

Reply:
I added a check higher up - basically any method with vararg will not get its return value inlined (regardless if it's static or not).